src/crimson/osd/osd.cc

```diff
@@ -123,16 +123,19 @@ seastar::future<> OSD::mkfs(uuid_d osd_uuid, uuid_d cluster_fsid)
       __func__,
       cluster_fsid,
       superblock.osd_fsid);
-
-    meta_coll = make_unique<OSDMeta>(
-      store->create_new_collection(coll_t::meta()), store.get());
-    ceph::os::Transaction t;
-    meta_coll->create(t);
-    meta_coll->store_superblock(t, superblock);
-    return store->do_transaction(meta_coll->collection(), std::move(t));
+    return store->create_new_collection(coll_t::meta()).then([this] (auto ch) {
```

Comment:
no need to embed this continuation. instead, we can just 1) put it into the next one 2) lift the next continuation. like
```c++
}).then([cluster_fsid, this] {
  return store->create_new_collection(coll_t::meta());
}).then([this](auto collection) {
  ceph::os::Transaction t;
  meta_coll = make_unique<OSDMeta>(collection , store.get());
  meta_coll->create(t);
  // ...
```